raft-tech/TANF-app
# Include all Cat2/3 Errors on Critical Tab of Error Report

## Summary of Changes
- Updated error report generation to accept a boolean indicating if it is a S3 or S4 file
- Updated the prioritized queryset function to prioritize the queryset differently for S3 and S4 files
- Exclude header update indicator cat2 errors wrt S3/S4 files
- Updated spreadsheet writer to exclude the case number column for S3/S4 files since it will always be empty
Pull request closes #3341 

~Something to note. S3/S4 records/errors do NOT have a case number association. However, the error report always includes a column on the critical tab for case number. In the S3/S4 case, that column will be completely empty. I have reached out to Alex to see if I can handle that case under this PR or if it should live in another ticket.~

## How to Test

```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Update a Aggregate/Stratum file to have some cat2/cat3 errors
3. Submit the file
4. Verify the report displays all cat2/cat3 errors on the "Critical" tab

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Any Cat 2 or Cat 3 errors detected in Aggregate data is included on critical worksheet
- [x] Any Cat 2 or Cat 3 errors detected in Stratum data is included on critical worksheet
- [x] The following cat 2 error message `does not` appear on critical worksheet: `HEADER Update Indicator must be set to D instead of N. Please review Exporting Complete Data Using FTANF in the Knowledge Center.`

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
